cli/ripsaw/clients/k8s.py

```diff
@@ -0,0 +1,169 @@
+        config.load_kube_config(config_file=kubeconfig_path)
+        self.api_client = client.ApiClient()
+        self.core_client = client.CoreV1Api()
+        self.batch_client = client.BatchV1Api()
+        self.crd_client = client.CustomObjectsApi()
+
+    def get_pods(self, label_selector, namespace):
+        return self.core_client.list_namespaced_pod(namespace, label_selector=label_selector, watch=False)
+
+    def get_pod_logs(self, label_selector, namespace, container):
+        pods = self.get_pods(label_selector, namespace).items
+        return [
+            self.core_client.read_namespaced_pod_log(
+                name=pod.metadata.name, namespace=namespace, container=container
+            )
+            for pod in pods
+        ]
+
+    def get_jobs(self, label_selector, namespace):
+        return self.batch_client.list_namespaced_job(namespace, label_selector=label_selector, watch=False)
+
+    def get_nodes(self, label_selector=None):
+        return self.core_client.list_node(label_selector=label_selector)
+
+    def get_node_names(self, label_selector=None):
+        return [node.metadata.name for node in self.get_nodes(label_selector).items]
+
+    def get_namespaces(self, label_selector=None):
+        return self.core_client.list_namespace(label_selector=label_selector)
+
+    def get_benchmark(self, name, namespace="benchmark-operator"):
+        return self.crd_client.get_namespaced_custom_object(
+            group="ripsaw.cloudbulldozer.io",
+            version="v1alpha1",
+            namespace=namespace,
+            plural="benchmarks",
+            name=name,
+        )
+
+    def get_benchmark_metadata(self, name, namespace="benchmark-operator"):
+        benchmark = self.get_benchmark(name, namespace)
+        return {
+            "name": benchmark["metadata"]["name"],
+            "namespace": benchmark["metadata"]["namespace"],
+            "uuid": benchmark.get("status", {}).get("uuid", "Not Assigned Yet"),
+            "suuid": benchmark.get("status", {}).get("suuid", "Not Assigned Yet"),
+            "status": benchmark.get("status", {}).get("state", ""),
+        }
+
+    # Waiters
+    def wait_for_pods(self, label_selector, namespace, timeout=300):
+        waiting_for_pods = True
+        timeout_interval = 0
+        while waiting_for_pods:
+            if timeout_interval >= timeout:
+                raise TimeoutError()
+            pods = self.get_pods(label_selector, namespace).items
+            if len(pods) == 0 and timeout_interval < 60:
+                continue
+
+            if len(pods) == 0 and timeout_interval > 60:
+                raise PodsNotFoundError(f"Found no pods with label selector {label_selector}")
+
+            _ = [
+                logger.info(f"{pod.metadata.namespace}\t{pod.metadata.name}\t{pod.status.phase}")
+                for pod in pods
+            ]
+            waiting_for_pods = any(  # pylint: disable=use-a-generator
```

Comment:
This should be an easy fix that doesn't require an ignore? By replacing the below brackets with parens?

```python
any((pod.status.phase != "Running" for pod in pods))
```

Reply:
fixed